lib/audit_help.c

```diff
@@ -68,6 +68,39 @@ void audit_logger (int type, MAYBE_UNUSED const char *pgname, const char *op,
 	}
 }
 
+/*
+ * This function will log a message to the audit system using a predefined
+ * message format. Parameter usage is as follows:
+ *
+ * type - type of message: AUDIT_USER_MGMT for changing any account
+ *	  attributes.
+ * pgname - program's name
+ * op  -  operation. "adding user", "changing finger info", "deleting group"
+ * name - user's account or group name. If not available use NULL.
+ * id  -  uid or gid that the operation is being performed on. This is used
+ *	  only when user is NULL.
+ * grp - group name associated with event
+ */
+void audit_logger_with_group (int type, MAYBE_UNUSED const char *pgname,
+		const char *op, const char *name, unsigned int id,
+		const char *grp, shadow_audit_result result)
+{
+	int len;
+	char enc_group[(GROUP_NAME_MAX_LENGTH*2)+1], buf[1024];
```

Comment:
That 1024 seems a bit arbitrary.  Could it maybe be less than sizeof(enc_group), which would be problematic?


Also,May we should do this?:

```c
char  enc_group[GROUP_NAME_MAX_LENGTH * 2 + 1];
char  buf[NITEMS(enc_group) + some value];  // How much would be enough?  Maybe 100?

```

(Also, the parentheses are redundant.)

Reply:
Sounds good to me. I've updated the code accordingly.